app/services/xapi/completed_product_service.rb

```diff
@@ -0,0 +1,29 @@
+module Xapi
+  class CompletedProductService < Xapi::BaseService
+    def initialize(agent, course)
+      @agent = agent
+      @object = object(course)
+    end
+
+    def set_statement
+      @statement = Xapi.create_statement(actor: @agent, verb: verb, object: @object)
+    end
+
+    private
+
+    def verb
+      Xapi.create_verb(id: 'http://adlnet.gov/expapi/verbs/completed', name: 'completed')
+    end
+
+    def object(course)
+      params = {
+          id: url_helpers.course_url(course, host: course.school.domains.primary.fqdn),
```

Comment:
I got confused by this `id` name. Maybe call it `resource`?

Reply:
Unfortunately, I think ID is part of the xAPI interface to identify the object resource.